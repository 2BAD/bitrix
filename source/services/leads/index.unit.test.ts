import { beforeEach, describe, expect, it, vi } from 'vitest'
import Leads from '.'

const mockCall = vi.fn(() => Promise.resolve()) as any
const mockList = vi.fn(() => Promise.resolve()) as any
const leads = Leads({ call: mockCall, list: mockList })

const ID = '77'

describe('Leads', () => {
  beforeEach(() => {
    mockCall.mockClear()
    mockList.mockClear()
  })

  it('should expose API methods', () => {
    expect(leads).toMatchSnapshot()
  })

  describe('`create`', () => {
    it('should invoke `call`', async () => {
      await leads.create({
        COMMENTS: 'Some comment'
      }, {
        REGISTER_SONET_EVENT: 'Y'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await leads.create({
        COMMENTS: 'Some comment'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`get`', () => {
    it('should invoke `call`', async () => {
      await leads.get(ID)
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })

  describe('`list`', () => {
    it('should invoke `list`', async () => {
      await leads.list({
        start: 0,
        order: { COMMENTS: 'ASC' },
        filter: { '>PROBABILITY': 50 },
        select: ['*', 'UF_*']
      })
      expect(mockList.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await leads.list()
      expect(mockList.mock.calls).toMatchSnapshot()
    })
  })

  describe('`update`', () => {
    it('should invoke `call`', async () => {
      await leads.update(ID, {
        COMMENTS: 'Some comment'
      }, {
        REGISTER_SONET_EVENT: 'Y'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })

    it('should work without optional params', async () => {
      await leads.update(ID, {
        COMMENTS: 'Some comment'
      })
      expect(mockCall.mock.calls).toMatchSnapshot()
    })
  })
})
