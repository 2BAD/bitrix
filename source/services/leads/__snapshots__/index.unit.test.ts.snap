// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leads \`create\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.lead.add",
    Object {
      "fields": Object {
        "COMMENTS": "Some comment",
      },
      "params": Object {
        "REGISTER_SONET_EVENT": "Y",
      },
    },
  ],
]
`;

exports[`Leads \`create\` should work without optional params 1`] = `
Array [
  Array [
    "crm.lead.add",
    Object {
      "fields": Object {
        "COMMENTS": "Some comment",
      },
      "params": undefined,
    },
  ],
]
`;

exports[`Leads \`get\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.lead.get",
    Object {
      "id": "77",
    },
  ],
]
`;

exports[`Leads \`list\` should invoke \`list\` 1`] = `
Array [
  Array [
    "crm.lead.list",
    Object {
      "filter": Object {
        ">PROBABILITY": 50,
      },
      "order": Object {
        "COMMENTS": "ASC",
      },
      "select": Array [
        "*",
        "UF_*",
      ],
      "start": 0,
    },
  ],
]
`;

exports[`Leads \`list\` should work without optional params 1`] = `
Array [
  Array [
    "crm.lead.list",
    Object {},
  ],
]
`;

exports[`Leads \`update\` should invoke \`call\` 1`] = `
Array [
  Array [
    "crm.lead.update",
    Object {
      "fields": Object {
        "COMMENTS": "Some comment",
      },
      "id": "77",
      "params": Object {
        "REGISTER_SONET_EVENT": "Y",
      },
    },
  ],
]
`;

exports[`Leads \`update\` should work without optional params 1`] = `
Array [
  Array [
    "crm.lead.update",
    Object {
      "fields": Object {
        "COMMENTS": "Some comment",
      },
      "id": "77",
      "params": undefined,
    },
  ],
]
`;

exports[`Leads should expose API methods 1`] = `
Object {
  "create": [Function],
  "fields": [Function],
  "get": [Function],
  "list": [Function],
  "update": [Function],
}
`;
